### ISSUE TITLE
Volumes of deleted pods do not get detached

### ISSUE THREAD

**Issue by dbaumgarten** [kubernetes/kubernetes#91771]
**What happened**: I have a stateful-set with 3 pods. Each pod has a persistent volume. When one of the pods is deleted (manually, by scaling down, via eviction etc.) the persistent volume of the pod stays attached to the node. Once the pod is re-surrected on another node, it gets a MultiAttachError, because the volume is still attached to the old node. After around 7 minutes the volume eventually gets force-detached from the old node and can now be successfully attached to the new node.

**What you expected to happen**: The persistent volume should be detached from the node as soon as the pod using it is deleted, so it is available for attachment once the deleted pod comes back on another node.

**How to reproduce it (as minimally and precisely as possible)**: Have a statefulset with persistent volumes on the pods. Delete one pod and have it re-created on another node. Observe that the volume takes around 7 minutes to finally be attached to the new node.

**Anything else we need to know?**: I have taken a look at the logs of the controllermanager. It does not even mention trying to (non-force) detach the volume. It only mentions,  MultiAttachErrors, Force-Detach and AttachSuccess.

**Environment**:
- Kubernetes version (use `kubectl version`): 1.16.9
- Cloud provider or hardware configuration: Openstack Ocata
- OS (e.g: `cat /etc/os-release`): Ubuntu 16.04.6
- Kernel (e.g. `uname -a`): 4.4.0-178-generic
- Install tools: kubeadm
- Others: I am using the internal (legacy) openstack cloud-provider

Cloud-config used:
```
[Global]
auth-url="********"
username="*********"
password="***********"
region="***********"
domain-id="default"
tenant-id="06bd1f27124e46f3938d2fef1f8551dc"
[LoadBalancer]
lb-provider="haproxy"
subnet-id="560b0339-39bd-4a43-8001-36c43e49007d"
create-monitor=true
monitor-delay=10s
monitor-timeout=3s
monitor-max-retries=1
floating-network-id="fe4fa7de-7649-444b-96d8-c64248d3f8f4"
[BlockStorage]
bs-version=v3
ignore-volume-az=true
```

**Comment by dbaumgarten** [comment]
I found out what is going wrong. 
I wanted to move the kubelet's root-dir to another volume, but as discussed[here](https://github.com/kubernetes/kubeadm/issues/1478) this is not that easy when you are using kubeadm.
This is why I  thought it was a good idea to bind-mount /var/lib/kubelet do another disk.
Unfortunately the kubelet does a lot of mount-juggeling inside it's root-dir and if that root-dir is a bind-mount itself weird things happen.

Because of the bind-mount weirdness kubelet was unable to un-mount volumes and while the volumes are still mounted they will not be detached from the node.

Tl;dr: Do not bind-mount /var/lib/kubelet to another location, or weird problems regarding volume (un-)mounts will arise.
